docs/src/pages/ui/primitives/passwordfield/react.mdx

```diff
@@ -0,0 +1,345 @@
+import {
+  Flex,
+  PasswordField,
+  TextField,
+  Button,
+  Heading,
+} from '@aws-amplify/ui-react';
+import {
+  PasswordFieldDemo,
+  LoginFormExample,
+  SignUpFormExample,
+  ChangePasswordFormExample,
+  DescriptiveTextExample,
+  RequiredPasswordFieldExample,
+  PasswordFieldStyledPropsExample,
+} from './demo';
+import { Example } from '@/components/Example';
+
+The PasswordField primitive can be used allow users to input passwords.
+
+## Demo
+
+<PasswordFieldDemo />
+
+## Usage
+
+Import the `PasswordField` component and styles and provide a `label` for accessibility/usability.
+
+```jsx
+import { PasswordField } from '@aws-amplify/ui-react';
+import '@aws-amplify/ui-react/styles.css';
+
+<PasswordField label="Password" name="password" />;
+```
+
+<Example>
+  <PasswordField label="Password" name="password" />
+</Example>
+
+### Hiding the show password button
+
+Use the `hideShowPassword` prop to hide the show password button.
+
+```jsx
+<PasswordField label="Password" name="password" hideShowPassword={true} />
+```
+
+<PasswordField label="Password" name="password" hideShowPassword={true} />
+
+### Autocomplete - supporting password managers
+
+Use the `autoComplete` prop to tell browser how to populate a password field. Options for `autoComplete` are `current-password` (default) and `new-password`.
+
+#### How to choose an `autoComplete` option
+
+If customers are logging in, you don't need to do set `autoComplete` because `current-password` is already the default.
+If customers are signing up for a new account, set `autoComplete` to `new-password`.
+
+See [MDN for more information on autocomplete](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete).
+
+_Signup form example: `new-password`_
+
+```jsx
+<form>
+  <Flex as="form" direction="column" gap="1rem">
+    <Heading level={3}>Sign Up</Heading>
+    <TextField label="Username" name="username" autoComplete="username" />
+    <PasswordField
+      label="Password"
+      name="password"
+      autoComplete="new-password"
+      descriptiveText="Password must be at least 8 characters"
+    />
+    <Button type="submit">Sign Up</Button>
+  </Flex>
+</form>
+```
+
+<Example>
+  <SignUpFormExample />
+</Example>
+
+_Login form example: `current-password` (default)_
+
+```jsx
+<Flex as="form" direction="column" gap="1rem">
+  <Heading level={3}>Login</Heading>
+  <TextField label="Username" name="username" autoComplete="username" />
+  <PasswordField
+    label="Password"
+    name="password"
+    autoComplete="current-password"
+  />
+  <Button type="submit" onClick={(e) => e.preventDefault()}>
+    Login
+  </Button>
+</Flex>
+```
+
+<Example>
+  <LoginFormExample />
+</Example>
+
+_Change password example: `current-password` and `new-password`_
+
+```jsx
```

Comment:
Just thinking out loud here... I wonder if there is a way we could keep the example code in the codeblock shown to users in sync with the actual example code? If we needed to make a change to this we would need to do it in 2 files. Not a huge deal or anything, just thoughts...

Reply:
Yeah, we need a good solution for this. I'll discuss with @slaymance , I think they solved this for the Authenticator docs.